src/ethereum/byzantium/vm/instructions/system.py

```diff
@@ -126,6 +126,7 @@ def create(evm: Evm) -> None:
         push(evm.stack, U256(0))
         evm.return_data = child_evm.output
     else:
+        evm.logs += child_evm.logs
```

Comment:
The last time I cleaned up this code,  I didn't want the `child_evm` logs to be a part of `evm.logs` so that there is no confusion while inspecting `evm` objects. 

Here's an alternate suggestion:
- Add a new attribute called `log_counter` to [Environment](https://github.com/ethereum/execution-specs/blob/a8ff48220f350decbba543015b4a2ea2b930aaf3/src/ethereum/frontier/vm/__init__.py#L29)
- Add a new attribute called `position` to [Log](https://github.com/ethereum/execution-specs/blob/a8ff48220f350decbba543015b4a2ea2b930aaf3/src/ethereum/frontier/eth_types.py#L134)
- Increment the `env.log_counter` every time a log is added and store this value in `Log`: `position = env.log_counter`
- sort the logs by `position` in [collect_logs](https://github.com/ethereum/execution-specs/blob/a8ff48220f350decbba543015b4a2ea2b930aaf3/src/ethereum/frontier/vm/interpreter.py#L289)

I can implement this myself if you'd like.

Reply:
I think the easiest thing is if you fix this in a separate PR.

Reply:
sure, I'll take care of this.

Reply:
I've been thinking about this a bit, and I'm not sure @voith's solution is the most spec appropriate.

The position/sort approach is what came to my mind as well when I was considering the bug, but it's less "obvious" than taking the logs from the child evm after exiting a call frame.

Would adding a `clear()` call after taking the logs sort out the confusion you mentioned @voith?

Reply:
Ah, interesting. So instead of it being tracked per call frame, you get one list per transaction.

How would that work if a frame is reverted?

Reply:
> How would that work if a frame is reverted?

Good question: logs should be cleared on Excepteptional halting:
https://github.com/ethereum/execution-specs/blob/2f35c4cdd8400990439ef53844dd487fbf7f20e0/src/ethereum/byzantium/vm/interpreter.py#L266-L269

Reply:
On second thought this won't work as I won't know which logs to clear when the frame reverts.

Reply:
Would it be possible to capture the message depth for each log within the log itself and then, during the final stage, remove the logs where depth is the same as an erred child evm?

Reply:
Well, that solution is not too different from what I suggested [here](https://github.com/ethereum/execution-specs/pull/473#discussion_r831425827). That's added complexity.

Reply:
Yeah. Plus I realised that there might be multiple EVMs at the same depth, so it won't work anyway cause we Weill end up deleting all the logs at a certain depth even if only one fails